### PR TITLE
[PVR] Timers: Allow to change timer type for existing timers.

### PR DESCRIPTION
**Pull request by ksooo**
This feature was requested several time sin the forum. With this PR it s possible in the timer settings dialog to edit the type of an _existing_ timer or timer rule, for instance to convert a one time timer to a series recording.

@Jalle19 mind taking a look at the code.

### CODE REVIEW

**Comment by Tolriq** [comment]
@Ksooo how does that impact JSON ? Does it need API change to allow that ?
I've seen your comment on a previous PR, is the naming rewrite planned for v17 ?

**Comment by ksooo** [comment]
@Tolriq no and no.

**Comment by Tolriq** [comment]
@ksooo Anyway to have the just the JSON chance for v17 ? 
There's already a lot of breaking  chance in JSON API for v17.
Timer API was added in JSON in v17 alpha.
Currently no one use that API.

That would simplify a lot the future if v17 and v18 are not incompatible on a new feature.

**Comment by ksooo** [comment]
Good point, not sure whether I find the time for this, though. PR welcome.

**Comment by Tolriq** [comment]
Tell me what to rename to what :) I have plans to include Timers in Yatse but for the moment I have no idea how it works as I do not use PVR in Kodi. So I do not even understand your naming changes :)

**Comment by ksooo** [comment]
"repeating timer" -> "timer rule". That's it.

**Comment by Tolriq** [comment]
Ok sounds pretty simple :)

Will try to find time before moving to do that.

**Comment by Tolriq** [comment]
@ksooo I've checked quickly current definition. Can you explain me what the new description should looks like for the setters ?

Currently it's "the add/delete/edit should be done for all occurances" for repeating, but I'm not sure to understand the difference of this new timer rule definition, so need your description as I suppose it's a little different behavior.

**Comment by ksooo** [comment]
Come up with a suggestion in a PR and I will be happy to review.

**Comment by ksooo** [comment]
@Jalle19 Good to go? 

@Tolriq thanks for the PR. Will have a look the next days.

**Comment by ksooo** [comment]
Ooops.

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp [hunk 1]
```diff
@@ -121,29 +121,38 @@ void CGUIDialogPVRTimerSettings::SetTimer(const CPVRTimerInfoTagPtr &timer)
   }
 
   // Copy data we need from tag. Do not modify the tag itself until Save()!
-  m_timerType     = m_timerInfoTag->GetTimerType();
+  m_timerType = m_timerInfoTag->GetTimerType();
+
+  if (!m_timerType)
```
Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: Are there really any situations where a timer tag could be created and stored without a timer type?

**Comment by Jalle19** [comment]
Apart from my previous comment, +1

File: xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: In theory that's possible, yes. But you may be right, it's just theory and of no real value here. I will drop this.

Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRTimerSettings.cpp]: Thanks

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
travis-ci errors are unrelated.

**Pull request merged**